src/com/google/javascript/jscomp/CodePrinter.java

```diff
@@ -340,6 +340,7 @@ void appendOp(String op, boolean binOp) {
         append(op);
         append(" ");
       } else {
+        append(" ");
```

Comment:
Would this be easier to read if you reformulated the conditions? Essentially you're now always prepending a whitespace, except for the lastchar/',' case (where op == ',' should always be a binOp, right?). So you could just check for the binOp case for the appended whitespace.